dialer.go

```diff
@@ -441,3 +435,23 @@ func splitHostPort(s string) (host string, port string) {
 	}
 	return
 }
+
+func runCustomResolver(ctx context.Context, address string, resolver Resolver) (string, error) {
```

Comment:
How about renaming to `lookupHost` since this is a helper to call `LookupHost` on the resolver?
```suggestion
func lookupHost(ctx context.Context, address string, resolver Resolver) (string, error) {
```

Reply:
I'm fine with this, the name I chose is a little awkward.